### PR TITLE
ATO-666: Re-verification result from IPV

### PR DESCRIPTION
**Pull request by isaac-GDS**
Add Base lambda
Add lambda infrastructure
Add api gateway endpoint on the fronted infrastructure

Policies and environment variables will need to be added as the lambda is developed.

### CODE REVIEW

File: ci/terraform/oidc/re-verification-result.tf [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  handler_function_name = "uk.gov.di.authentication.frontendapi.lambda.ReverificationResultHandler::handleRequest"
+
+  rest_api_id      = aws_api_gateway_rest_api.di_authentication_frontend_api.id
+  root_resource_id = aws_api_gateway_rest_api.di_authentication_frontend_api.root_resource_id
+  execution_arn    = aws_api_gateway_rest_api.di_authentication_frontend_api.execution_arn
+
+  memory_size                 = lookup(var.performance_tuning, "re-verification-result", local.default_performance_parameters).memory
+  provisioned_concurrency     = lookup(var.performance_tuning, "re-verification-result", local.default_performance_parameters).concurrency
```
Comment (ethanmills) [inline ci/terraform/oidc/re-verification-result.tf:30]: We can set the provisioned concurrence to 1 on this where it defaults to 3. It's going to be very low volume I expect

File: ci/terraform/oidc/re-verification-result.tf [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+  handler_function_name = "uk.gov.di.authentication.frontendapi.lambda.ReverificationResultHandler::handleRequest"
+
+  rest_api_id      = aws_api_gateway_rest_api.di_authentication_frontend_api.id
+  root_resource_id = aws_api_gateway_rest_api.di_authentication_frontend_api.root_resource_id
+  execution_arn    = aws_api_gateway_rest_api.di_authentication_frontend_api.execution_arn
+
+  memory_size                 = lookup(var.performance_tuning, "re-verification-result", local.default_performance_parameters).memory
+  provisioned_concurrency     = 1
```
Comment (ethanmills) [inline ci/terraform/oidc/re-verification-result.tf:30]: Ah sorry, this is not what I meant. We need to set the values in the environment files. There should still be no provisioned concurrency in dev, like the default

**Review by ethanmills — COMMENTED** [review]
Sorry I've only just spotted this, but I think it should be reverification, not re-verification, to match the IPV endpoint and the scope name

**Comment by isaac-GDS** [comment]
> Sorry I've only just spotted this, but I think it should be reverification, not re-verification, to match the IPV endpoint and the scope name

Ah nice - yeah shall do. Good spot.

**Review by ethanmills — DISMISSED** [review]

**Review by ethanmills — APPROVED** [review]

**Pull request merged**